Cargo.toml

```diff
@@ -15,6 +15,7 @@ build = "build.rs"
 [dependencies]
 regex = "0.1"
 lazy_static = "0.1.15"
+itertools = "0.4.15"
```

Comment:
Sorry, I was traveling. You could solve it like this:
```rust
let mut result = strings_to_join[0].to_string();
for string in strings_to_join.iter().skip(1).map(|x| x.as_str()) {
    result.push_str(join_string);
    result.push_str(string);
}
Ok(Str(result))
```
However, you can shorten the code above (where you declare `strings_to_join`) to simply:

```rust
let mut strings_to_join = array.iter().map(|x| x.to_string());
```

which enables us to simplify the loop below a lot:

```rust
let mut result = strings_to_join.next().unwrap_or(String::new());
for string in strings_to_join {
         result.push_str(&join_string);
         result.push_str(&string);
}
Ok(Str(result))
```
And that allows us to avoid a lot of type conversion and solves it in a single iteration of `array`.